File: langhuan-service/src/main/java/com/langhuan/advisors/MySimpleLoggerAdvisor.java
package com.langhuan.advisors;

import lombok.extern.slf4j.Slf4j;

import org.springframework.ai.chat.client.ChatClientMessageAggregator;
import org.springframework.ai.chat.client.ChatClientRequest;
import org.springframework.ai.chat.client.ChatClientResponse;
import org.springframework.ai.chat.client.advisor.api.*;
import org.springframework.ai.chat.model.MessageAggregator;
import reactor.core.publisher.Flux;
import java.util.function.Function;

@Slf4j
public class MySimpleLoggerAdvisor implements CallAdvisor, StreamAdvisor {

	@Override
	public String getName() {
		return this.getClass().getSimpleName();
	}

	@Override
	public int getOrder() {
		return 0;
	}

	@Override
	public Flux<ChatClientResponse> adviseStream(ChatClientRequest chatClientRequest,
			StreamAdvisorChain streamAdvisorChain) {
		this.logRequest(chatClientRequest);
		Flux<ChatClientResponse> chatClientResponses = streamAdvisorChain.nextStream(chatClientRequest);
		return (new ChatClientMessageAggregator()).aggregateChatClientResponse(chatClientResponses, this::logResponse);
	}

	@Override
	public ChatClientResponse adviseCall(ChatClientRequest chatClientRequest, CallAdvisorChain callAdvisorChain) {
		this.logRequest(chatClientRequest);
		ChatClientResponse chatClientResponse = callAdvisorChain.nextCall(chatClientRequest);
		this.logResponse(chatClientResponse);
		return chatClientResponse;
	}

	private void logRequest(ChatClientRequest request) {
		log.debug("request: {}", request);
	}

	private void logResponse(ChatClientResponse chatClientResponse) {
		log.debug("response: {}", chatClientResponse.chatResponse());
	}

}

File: langhuan-server/src/main/kotlin/com/langhuan/advisors/MySimpleLoggerAdvisor.kt
package com.langhuan.advisors

import org.slf4j.LoggerFactory
import org.springframework.ai.chat.client.ChatClientMessageAggregator
import org.springframework.ai.chat.client.ChatClientRequest
import org.springframework.ai.chat.client.ChatClientResponse
import org.springframework.ai.chat.client.advisor.api.CallAdvisor
import org.springframework.ai.chat.client.advisor.api.CallAdvisorChain
import org.springframework.ai.chat.client.advisor.api.StreamAdvisor
import org.springframework.ai.chat.client.advisor.api.StreamAdvisorChain
import reactor.core.publisher.Flux

class MySimpleLoggerAdvisor : CallAdvisor, StreamAdvisor {

    override fun getName(): String {
        return this::class.java.simpleName
    }

    override fun getOrder(): Int {
        return 0
    }

    override fun adviseStream(
        chatClientRequest: ChatClientRequest,
        streamAdvisorChain: StreamAdvisorChain
    ): Flux<ChatClientResponse> {
        logRequest(chatClientRequest)
        val chatClientResponses = streamAdvisorChain.nextStream(chatClientRequest)
        return ChatClientMessageAggregator().aggregateChatClientResponse(chatClientResponses, this::logResponse)
    }

    override fun adviseCall(
        chatClientRequest: ChatClientRequest,
        callAdvisorChain: CallAdvisorChain
    ): ChatClientResponse {
        logRequest(chatClientRequest)
        val chatClientResponse = callAdvisorChain.nextCall(chatClientRequest)
        logResponse(chatClientResponse)
        return chatClientResponse
    }

    private fun logRequest(request: ChatClientRequest) {
        log.debug("request: {}", request)
    }

    private fun logResponse(chatClientResponse: ChatClientResponse) {
        log.debug("response: {}", chatClientResponse.chatResponse())
    }

    companion object {
        private val log = LoggerFactory.getLogger(MySimpleLoggerAdvisor::class.java)
    }
}
